CMakeLists.txt

```diff
@@ -46,8 +46,9 @@ macro(build_spdlog)
     list(APPEND extra_cmake_args "-DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}")
   endif()
   list(APPEND extra_cmake_args "-DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}")
-  list(APPEND extra_cmake_args "-DSPDLOG_BUILD_TESTS=OFF")
+  list(APPEND extra_cmake_args "-DSPDLOG_BUILD_TESTS=ON")
   list(APPEND extra_cmake_args "-DSPDLOG_BUILD_EXAMPLE=OFF")
+  list(APPEND extra_cmake_args "-DSPDLOG_BUILD_BENCH=ON")
```

Comment:
I'm don't think we should turn on the tests and benchmarks for the vendored package.  First of all, are we going to run them?  Second, even if we can run them, should we run them?  I guess you are running them down below, so the answer to the first is "yes".  But I don't really think we should run them, to be honest.

Reply:
I believe the goal of running these tests is to detect regressions that could affect our code. It looks like we're pinning a specific commit, so maybe the risk that something breaks is very low. Is there a chance that something breaks due to factors external to the source code?

What's the concern with enabling tests, added time on our CI?

If one thing, I think the tests on this repository should respect the `BUILD_TESTING` variable.

Reply:
We have done this for console_bridge_vendor to reach queality level 1. spdlog is running the test in Travis, but the benchmark tests are not running anywhere.

 - We are using a specific version of the code. Maybe it does not make sense to add any benchmark because the code it't not going to change
 - But at least we have all the benchmark in the same place.

Same logic applies for the API tests.

Reply:
> What's the concern with enabling tests, added time on our CI?

That's one concern.  Another is that we can't build and run tests for every single thing we depend on.  Some things just come from the system libraries, and we assume they are "good" enough.

> We have done this for console_bridge_vendor to reach queality level 1.

I didn't realize that.  I do think that it is in somewhat of a similar situation in that we shouldn't necessarily build the tests for it.  But on the other hand, console_bridge is special in that we also maintain it, just not as a ROS 2 package.

In the end, if you think it adds value to run the tests here, then go ahead and keep it.  I just wanted to question why we were doing this, and whether we should.  But I agree with @chapulina that it should definitely respect `BUILD_TESTING`.

Reply:
> we can't build and run tests for every single thing we depend on

Yeah the hope is that the upstream projects themselves would be tested to a level that we consider satisfying for REP-2004. Unfortunately this isn't the case for spdlog, so one way of satisfying REP-2004 is by testing at least the parts that we care about. Not ideal, but I can't think of an alternative solution... :confused: